tests/integration/istioio/trafficmanagement/scripts/request_timeouts.txt

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# TODO: Find better way to re-use HOST in multiple test
+# Export HOST, PORT and URL
+# $snippet export_gateway_url_1.sh syntax="bash"
+INGRESS_NS=${INGRESS_NS:-istio-system}
+INGRESS_HOST=$(kubectl -n "${INGRESS_NS}" get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+export INGRESS_HOST
+INGRESS_PORT=$(kubectl -n "${INGRESS_NS}" get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+export INGRESS_PORT
+SECURE_INGRESS_PORT=$(kubectl -n "${INGRESS_NS}" get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+export SECURE_INGRESS_PORT
+if [ -z "$INGRESS_HOST" ]; then
+    INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n "${INGRESS_NS}" -o jsonpath='{.items[0].status.hostIP}')
+    export INGRESS_HOST
+    INGRESS_PORT=$(kubectl -n "${INGRESS_NS}" get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+    export INGRESS_PORT
+    SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+    export SECURE_INGRESS_PORT
+fi
+MINIKUBE_IP=$(minikube ip)
```

Comment:
this doesn't work if minikube isn't present :)

Reply:
This looks like a better way to handle the ingress host/port.  

https://github.com/su225/istio/blob/18285-tcp-traffic-shifting/tests/integration/istioio/trafficmanagement/trafficshifting/tcp_trafficshifting_test.go#L48..L61

Reply:
The link you shared was broken, but I found ref in [SDS test](https://github.com/istio/istio/blob/d32a4972c9aad6586d141d4a80b204f51ba6d073/tests/integration/istioio/trafficmanagement/ingress/secure_ingress_sds_test.go#L58)

Thanks for sharing the link. This is definitely better way to handle host and port, it also helps passing context so we can re-use `HOST/PORT` in multiple files.

Reply:
I think I'm missing some configs because the test is not able to find host. I'm getting following error.
```go
--- FAIL: TestRequestTimeouts (33.31s)
    script.go:364: script scripts/request_timeouts.txt returned an error: exit status 1. Output:
        virtualservice.networking.istio.io/reviews configured
        virtualservice.networking.istio.io/ratings configured
        sending request to :31400
        curl: (6) Could not resolve host: 
FAIL
FAIL    istio.io/istio/tests/integration/istioio/trafficmanagement      198.690s
```
I'm using following command to run single test.
```go
go test -p 1  ./tests/integration/istioio/trafficmanagement --istio.test.work_dir /home/shaansar/Projects/src/istio.io/istio/out/linux_amd64  -timeout 30m \
--istio.test.select -postsubmit,-flaky \
--istio.test.env kube \
--istio.test.kube.config ~/.kube/config \
--istio.test.hub=docker.io/istio \
--istio.test.tag=1.5.0-alpha.0 \
--istio.test.pullpolicy=IfNotPresent
```
Let me know if I miss something.

Reply:
@brian-avery @howardjohn can you please have a look and let me know if I'm doing something wrong with my configs?